Firestore/core/src/firebase/firestore/util/comparison.h

```diff
@@ -353,6 +353,35 @@ class Comparable {
   }
 };
 
+/**
+ * Same as `Comparable`, but `operator==` is delegated to a function called
+ * `IsEqualTo`. This is an optimization that avoids doing an extra comparison
+ * when comparing for equality (`Comparable` would have to check for both
+ * "less-than" and "greater-than" to determine whether two values are equal).
+ */
+template <typename T>
+class ComparableWithEqual {
```

Comment:
One alternative is to only have `Comparable`, use `enable_if` to switch based on whether `T` defines `IsEqualTo`. Seems overkill (and more error prone, because forgetting to define `IsEqualTo` will not be an error).

Reply:
That doesn’t seem warranted for just the handful of cases we have here. `std::enable_if` should be an absolute last resort.